src/model/labels.js

```diff
@@ -0,0 +1,50 @@
+const _ = require('lodash');
+const Label = require('./label');
+
+module.exports.build = function(labels) {
+  return labels.map(label => new Label(label));
+};
+
+module.exports.compare = function(currentLabels, futureLabels) {
+  let toCreate = _.differenceBy(futureLabels, currentLabels, 'name');
+  let toDelete = _.differenceBy(currentLabels, futureLabels, 'name');
+
+  const toUpdate = [];
+
+  futureLabels.forEach(newLabel => {
+    const matching = _.filter(currentLabels, ['name', newLabel.name]);
+
+    matching.forEach(matchingLabel => {
+      if (matchingLabel.hash() !== newLabel.hash()) {
+        toUpdate.push({
+          oldLabel: matchingLabel,
+          newLabel,
+        });
+      }
+    });
+
+    if (matching.length !== 0) {
```

Comment:
Should we check this directly after line 15 so we can exit early?

Reply:
nope, cause the method above this adds things to the toUpdate array, so we need to execute that one.

Reply:
Oh yeah, you are right! Sorry I misread that as `matching.length === 0`